packages/core/src/__tests__/ceramic-recover-streams.test.ts

```diff
@@ -80,3 +97,21 @@ it("re-request anchors on #recoverStreams", async () => {
     expect(stream2.state.anchorStatus).toEqual(AnchorStatus.ANCHORED);
     await ceramic2.close();
 });
+
+it("Don't create new anchor request on #recoverStreams", async () => {
```

Comment:
So although this test passes, it logs some warnings I find worrisome:

```
  console.log
    Anchor failed for commit bagcqcerabkhoyxbs6bpuue767lf5neny3j5url4neynpltjii2eceyngauia of stream kjzl6cwe1jw145jcta3t0c8abrjj9acnmnks10dv7qoxai6w9qtgz3ulhccixao: Request not found

      at Function.Object.<anonymous>.Logger.PrintLog (../../node_modules/@overnightjs/logger/lib/Logger.js:173:21)

  console.log
    Error while anchoring stream kjzl6cwe1jw145jcta3t0c8abrjj9acnmnks10dv7qoxai6w9qtgz3ulhccixao:ReferenceError: Cannot access 'subscription' before initialization

      at Function.Object.<anonymous>.Logger.PrintLog (../../node_modules/@overnightjs/logger/lib/Logger.js:173:21)

  console.log
    ReferenceError: Cannot access 'subscription' before initialization
        at /home/spencer/js-ceramic2/packages/core/src/state-management/state-manager.ts:278:15
        at doInnerSub (/home/spencer/js-ceramic2/node_modules/rxjs/src/internal/operators/mergeInternals.ts:70:15)
        at outerNext (/home/spencer/js-ceramic2/node_modules/rxjs/src/internal/operators/mergeInternals.ts:52:58)
        at OperatorSubscriber._this._next (/home/spencer/js-ceramic2/node_modules/rxjs/src/internal/operators/OperatorSubscriber.ts:43:13)
        at OperatorSubscriber.Object.<anonymous>.Subscriber.next (/home/spencer/js-ceramic2/node_modules/rxjs/src/internal/Subscriber.ts:74:12)
        at /home/spencer/js-ceramic2/node_modules/rxjs/src/internal/operators/mergeInternals.ts:84:24
        at OperatorSubscriber._this._next (/home/spencer/js-ceramic2/node_modules/rxjs/src/internal/operators/OperatorSubscriber.ts:43:13)
        at OperatorSubscriber.Object.<anonymous>.Subscriber.next (/home/spencer/js-ceramic2/node_modules/rxjs/src/internal/Subscriber.ts:74:12)
        at Observable._subscribe (/home/spencer/js-ceramic2/node_modules/rxjs/src/internal/observable/from.ts:187:18)
        at Observable.Object.<anonymous>.Observable._trySubscribe (/home/spencer/js-ceramic2/node_modules/rxjs/src/internal/Observable.ts:287:19)

      at Function.Object.<anonymous>.Logger.PrintLog (../../node_modules/@overnightjs/logger/lib/Logger.js:173:21)
```

The first message is expected, but the second 2 are not.

This seems to come from the `subscription.unsubscribe()` call in the `FAILED` case of `_processAnchorResponse`.  I guess because we return a FAILED message immediately, as the first thing the Observable emits, the `.subscribe()` call at the end of `_processAnchorResponse` hasn't ever run yet.  So I guess the way the code is structured now the Observable needs to emit at least one event before emitting FAILED or ANCHORED.  That seems like a potential problem, but I'm not sure how to restructure things to fix it.  Would be interested if you have any insight here @ukstv

Reply:
This seems like a potential problem indeed. Will do a PR to this PR first thing in the morning.

Reply:
Or, actually, here is a quick demonstration of the approach before the PR (and if it is needed): https://gist.github.com/ukstv/61b7028b3eefb708c826ef2b401bdd7e

Reply:
Okay that makes sense, thanks.  I can take a crack at making that change

Reply:
Okay, updated!